client/lib/signup/step-actions.js

```diff
@@ -415,7 +415,24 @@ export function createAccount(
 					);
 				}
 
-				analytics.recordSocialRegistration();
+				const userId =
+					( response && response.signup_sandbox_user_id ) ||
+					( response && response.user_id ) ||
+					userData.ID;
+
+				const username =
+					( response && response.signup_sandbox_username ) ||
+					( response && response.username ) ||
+					userData.username;
+
+				const registrationUserData = {
+					ID: userId,
+					username: username,
+					email: userData.email
```

Comment:
This throws an error in social signup for me (tested in a local dev environment using the `Continue with Google` button). It looks like for [Google signups](https://github.com/automattic/wp-calypso/blob/f2c9d99b07e14ea938829e9b2e5b31b6fb388a90/client/blocks/signup-form/social.jsx#L55) no additional userData is passed in so this will be `null`. And for [Apple signups](https://github.com/automattic/wp-calypso/blob/f2c9d99b07e14ea938829e9b2e5b31b6fb388a90/client/blocks/signup-form/social.jsx#L40), the property is `user_email` instead of `email` — I haven't had a chance to test signing in with Apple, but I suspect this will throw the same error. Perhaps the Apple one could be renamed? For Google, it looks like there should be an email address in the response object at `response.email`, I think?

Reply:
Error message was `step-actions.js:421 Uncaught TypeError: Cannot read property 'email' of null`